### PR TITLE
Sudoku example modified

### PR DESCRIPTION
**Pull request by spiralulam**
Dear Stu,

as discussed via email I modified the two sudoku example files and changed the following:

- Streamline the whole code
- Removing some constraints from the second sudoku file in order to obtain a sudoku with multiple solutions

Best,

Nathan

### CODE REVIEW

File: examples/Sudoku1.py [hunk 1]
```diff
@@ -1,81 +1,76 @@
...
-prob += choices["9"]["8"]["6"] == 1,""
-prob += choices["2"]["7"]["7"] == 1,""
-prob += choices["6"]["3"]["8"] == 1,""
-prob += choices["8"]["7"]["8"] == 1,""
-prob += choices["7"]["9"]["8"] == 1,""
-prob += choices["3"]["4"]["9"] == 1,""
-prob += choices["1"]["5"]["9"] == 1,""
-prob += choices["6"]["6"]["9"] == 1,""
-prob += choices["5"]["8"]["9"] == 1,""
+prob += choices[5][1][1] == 1
```
Comment (pchtsp) [inline examples/Sudoku1.py]: Hey!

Just a quick suggestion. Would it be desirable to do something like:

```python
input_data = [
(6, 2, 1), 
(8, 4, 1), 
(4, 5, 1)
]

for (v, r, c) in input_data:
	prob += choices[v][r][c] == 1

```

File: examples/Sudoku1.py [hunk 2]
```diff
@@ -90,19 +85,16 @@
 sudokuout = open('sudokuout.txt','w')
 
 # The solution is written to the sudokuout.txt file 
-for r in Rows:
-    if r == "1" or r == "4" or r == "7":
+for r in ROWS:
+    if r == 1 or r == 4 or r == 7:
```
Comment (pchtsp) [inline examples/Sudoku1.py]: not really that important but... 

```
r in [1, 4, 7]:

```

may look nicer?

File: examples/Sudoku1.py (hunk 1 above)
Comment (spiralulam) [reply examples/Sudoku1.py]: Yeah, for sure, that is much nicer.

File: examples/Sudoku1.py (hunk 2 above)
Comment (spiralulam) [reply examples/Sudoku1.py]: Agreed!

offtopic: This is my first pull request, so I don't know what is best practice here. Do I have to create a new pull request that incorporates your suggestions?

Comment (pchtsp) [reply examples/Sudoku1.py]: You don't need to create another Pull Request as long as you don't close this one. Just do the necessary commits in the same branch and push your changes. They will appear in this PR.

Comment (spiralulam) [reply examples/Sudoku1.py]: Thx, I submitted the changes.

**Comment by spiralulam** [comment]
I just recognized that I also have to adjust the documentation here 

https://github.com/coin-or/pulp/blob/master/doc/source/CaseStudies/a_sudoku_problem.rst

I will do that within the next week

**Comment by spiralulam** [comment]
Ok, done.

**Comment by spiralulam** [comment]
Stu, is something missing here? Best, Nathan

**Comment by pchtsp** [comment]
Hello again! can you please solve the conflicts so we can merge the branch? Thanks!

**Comment by spiralulam** [comment]
Hi, done! :-)

**Pull request merged**